Eatery/Controllers/Eateries/EateriesViewController.swift

```diff
@@ -398,13 +410,32 @@ class EateriesViewController: UIViewController {
                                                                sortMethodWithSearchText: searchText,
                                                                filters: filterBar.selectedFilters)
 
-            eateriesByGroup = eateriesByGroup(from: eateries, sortedUsing: sortMethod)
+            newEateriesByGroup = eateriesByGroup(from: eateries, sortedUsing: sortMethod)
         } else {
-            eateriesByGroup = (favorites: [], open: [], closed: [])
+            newEateriesByGroup = (favorites: [], open: [], closed: [])
         }
 
-        UIView.performWithoutAnimation {
+        if let oldEateriesByGroup = eateriesByGroup {
+            let oldEateriesById = (oldEateriesByGroup.favorites.map { $0.id },
+                                   oldEateriesByGroup.open.map { $0.id },
+                                   oldEateriesByGroup.closed.map { $0.id })
+            let newEateriesById = (newEateriesByGroup.favorites.map { $0.id },
+                                   newEateriesByGroup.open.map { $0.id },
+                                   newEateriesByGroup.closed.map { $0.id })
+
+            if oldEateriesById == newEateriesById {
+                return
+            }
+        }
+
+        self.eateriesByGroup = newEateriesByGroup
+
+        if animated {
             collectionView.reloadSections(IndexSet(1...3))
+        } else {
+            UIView.performWithoutAnimation {
+                collectionView.reloadSections(IndexSet(1...3))
```

Comment:
maybe extract `IndexSet(1...3)` into a variable so that its clear what it represents and also because we use it twice